Sources/Dollar.swift

```diff
@@ -561,6 +561,23 @@ public class $ {
         return (first / $.gcd(first, second)) * second
     }
     
+    /// Splits a collection into sets, grouped by the result of running each value through a callback.
+    ///
+    /// :param The array to group
+    /// :param callback Function whose response will be used as a key in the new string
+    public class func groupBy<T, U>(array: [T], callback: (T) -> U) -> [String: [T]]{
```

Comment:
Why are you using String. Can we make the dictionary type be `[Hashable: [T]]` so it works with a lot of other types.